plugins/package-managers/gleam/src/main/kotlin/Gleam.kt

```diff
@@ -73,7 +73,10 @@ class Gleam internal constructor(
     ) {
         super.beforeResolution(analysisRoot, definitionFiles, analyzerConfig)
 
-        projectDirs = definitionFiles.mapTo(mutableSetOf()) { it.parentFile.canonicalFile }
+        projectDirs.apply {
+            clear()
+            definitionFiles.mapTo(this) { it.parentFile.canonicalFile }
```

Comment:
Actually, by now I don't see a reason for this being done in `beforeResolution()` at all. Why not do it directly in `resolveDependencies()`?

Reply:
This needs to have all (mapped) definition files, while `resolveDependencies()` only has a single one.

Reply:
I see. However, as there is a guarantee that an instance of a package manager will only be used once, and `beforeResolution()` will always be called before `resolveDependencies()`, how about instead changing line 67 to

```kotlin
private lateinit var projectDirs: Set<File>
```